pkg/cache/cache.go

```diff
@@ -221,6 +238,16 @@ func (c *Cache) ClusterQueueTerminating(name string) bool {
 	return c.clusterQueueInStatus(name, terminating)
 }
 
+func (c *Cache) ClusterQueueInactiveReason(name string) (string, string) {
+	c.RLock()
+	defer c.RUnlock()
+	cq, exists := c.clusterQueues[name]
+	if !exists {
+		return "NotFound", "Cluster queue not found"
+	}
+	return cq.inactiveReason()
+}
```

Comment:
Keeping the work shorter is the reason why I've done it in cache, otherwise the queue controller should do multiple calls (with locks) to get the same info.

Reply:
While I sympathize with the idea that this should be controller code, I agree with Traian, obtaining the lock once will be more performant.

Reply:
> Keeping the work shorter is the reason why I've done it in cache, otherwise the queue controller should do multiple calls (with locks) to get the same info.

It sounds reasonable.

> While I sympathize with the idea that this should be controller code, I agree with Traian, obtaining the lock once will be more performant.

Probably, we need to refactor the cache and then remove unneeded information from the cache in other PRs, although, we hold the information inside the cache in this PR 😞